src/setup_payload/QRCodeSetupPayloadParser.h

```diff
@@ -43,14 +44,26 @@ class QRCodeSetupPayloadParser
     std::string mBase38Representation;
 
 public:
+    /**
+     * base38Representation is expected to start with the "MT:" prefix, in general.
+     */
     QRCodeSetupPayloadParser(std::string base38Representation) : mBase38Representation(std::move(base38Representation)) {}
     CHIP_ERROR populatePayload(SetupPayload & outPayload);
     static std::string ExtractPayload(std::string inString);
 
+    /**
+     * Parse the provided representation into a vector of payloads, to handle concatenated payloads.
+     *
+     * Returns error if there are errors, else the resulting vector.
+     */
+    std::variant<CHIP_ERROR, std::vector<SetupPayload>> Parse();
```

Comment:
The `populatePayload` vs `Parse` naming seems a bit incongruous given how similar the two methods are. Maybe this one could be called `ParsePayloads` to make them a bit more similar. I would also be nice to document which one to use when.

Reply:
Well, this Parse() thing works even if there is only one payload.  But yes, I can do that.

Reply:
Done.